### PR TITLE
HDDS-2294. Create a new HISTORY.md in the new repository.

### PR DESCRIPTION
**Pull request by elek**
## What changes were proposed in this pull request?

During the apache/hadoop.git --> apache/hadoop-ozone.git move we rewrote the (git) history to simplify the work. Unfortunately some of the early work of HDDS-7280 is not part of the hadoop-ozone repository just the hadoop repository. As it suggested by Anu Engineer, we can explain this in a separated file and show how the origin of Ozone can be found.

*Note*: I wrote it after the repository separation but forgot to create the PR. I just found this doc in my documents folder and found it today.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2294

## How was this patch tested?

Read it, please.

### CODE REVIEW

File: HISTORY.txt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+# History of Apache Hadoop Ozone project
+
+Ozone development is started on a feature branch of HDFS-7240 as part of the Apache Hadoop HDFS project. Based on the Jira information the first Ozone commit was the commit of [HDFS-8456 Ozone: Introduce STORAGE_CONTAINER_SERVICE as a new NodeType.](https://issues.apache.org/jira/browse/HDFS-8456) in the May of 2015.
```
[the remark this answers is not in the document]
Comment (dineshchitlangia) [reply HISTORY.txt:20]: Yeah, used it a few times, this time I am writing this from phone while waiting for my grocery pick up 🤣

File: HISTORY.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Ozone is an Object Store for Hadoop which is based on a lower level storage replication layer. This layer is called originally HDSL (Hadoop Distributed Storage Layer) and renamed later to HDDS (Hadoop Distributed Data Storage).
+
+This generic storage layer started be implemented under [HDFS-11118](https://issues.apache.org/jira/browse/HDFS-11118) together with a jScsi based block storage layer ("CBlock") introduced by [HDFS-11361](https://issues.apache.org/jira/browse/HDFS-11361).
```
[the remark this answers is not in the document]
Comment (elek) [reply HISTORY.txt:24]: It was based on this project https://github.com/sebastiangraf/jSCSI and the project maintainers helped a lot to quickly fix the problem.

I fixed it to iScsi/jScsi (with the link to the jScsi project)

File: HISTORY.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+As the CBlock feature was not stable enough it was not merged and archived on a separated feature branch which was not synced with the newer Ozone/HDDS features. (Somewhat similar functionality is provided later with S3 Fuse file system and an S3 compatible REST gateway.)
+
+After the merge a new Jira project was created (HDDS) and the work was tracked under that project instead of in the child issues of HDFS-7240 issue.
+
+In the next year multiple Ozone releases has been published in separated release package. The Ozone source release was developed on the Hadoop trunk, but the Ozone sources are removed from the main Hadoop releases.
+
+Originally Ozone depended on the in-tree (SNAPSHOT) Hadoop artifacts. It was required to compile the core hadoop-hdfs/hadoop-common artifacts before compiling the Ozone subprojects. During the development this dependency is reduced more and more. With the 0.4.1 release this dependency has bee totally removed and it became possible to compile Ozone with the help of the released Hadoop artifacts which made it possible to separate the development of Ozone from the main Hadoop trunk branch.
```
[the remark this answers is not in the document]
Comment (elek) [reply HISTORY.txt:51]: > Typo: dependency has bee -> dependency has been

:honeybee:

File: HISTORY.txt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+In the October of 2017 a [discussion](https://lists.apache.org/thread.html/3b5b65ce428f88299e6cb4c5d745ec65917490be9e417d361cc08d7e@%3Chdfs-dev.hadoop.apache.org%3E) has been started on hdfs-dev mailing list to merge the existing functionality to the Apache Hadoop trunk. After a long debate Owen O'Malley [suggested a consensus](https://lists.apache.org/thread.html/c85e5263dcc0ca1d13cbbe3bcfb53236784a39111b8c353f60582eb4@%3Chdfs-dev.hadoop.apache.org%3E) to merge it to the trunk but use separated release cycle:
+
+ > * HDSL become a subproject of Hadoop.
+ > * HDSL will release separately from Hadoop. Hadoop releases will notcontain HDSL and vice versa.
+ > * HDSL will get its own jira instance so that the release tags stay separate.
+ > * On trunk (as opposed to release branches) HDSL will be a separate module in Hadoop's source tree. This will enable the HDSL to work on their trunk and the Hadoop trunk without making releases for every change.
+ > * Hadoop's trunk will only build HDSL if a non-default profile is enabled. When Hadoop creates a release branch, the RM will delete the HDSL module from the branch.
```
[the remark this answers is not in the document]
Comment (elek) [reply HISTORY.txt:40]: This is the text of the origin mail from Owen (`>` should be rendered as quote). I would leave this sentence unchanged.

https://lists.apache.org/thread.html/c85e5263dcc0ca1d13cbbe3bcfb53236784a39111b8c353f60582eb4@%3Chdfs-dev.hadoop.apache.org%3E

**Comment by elek** [comment]
Thank you very much the proofreading @dineshchitlangia (and @adoroszlai). It's a big help for me.

I have to answers, all the other suggestions are applied.

File: HISTORY.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (dineshchitlangia) [reply HISTORY.txt:40]: Sounds good.

File: HISTORY.txt [hunk 5]
```diff
@@ -0,0 +1,60 @@
+<!---
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply HISTORY.txt:1]: @elek did you intend to rename it?  I still see `.txt`.

File: HISTORY.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+Implementation of the generic storage layer began under [HDFS-11118](https://issues.apache.org/jira/browse/HDFS-11118) together with a iScsi/[jScsi](https://github.com/sebastiangraf/jSCSI) based block storage layer ("CBlock") introduced by [HDFS-11361](https://issues.apache.org/jira/browse/HDFS-11361).
+
+As a summary:
+
+ * HDDS (earlier HDSL): replicates huge binary _containers_ between datanodes
+ * Ozone: provides Object Store semantics with the help of HDDS
+ * CBlock: provides mountable volumes with the help of the HDDS layer (based on iScsi protocol)
+
+In the beginning of the year 2017 a new poddling project was started inside [Apache Incubator](http://incubator.apache.org/): [Apache Ratis](https://ratis.apache.org/). Ratis is an embeddable RAFT protcol implementation it is which became the corner stone of consensus inside both Ozone and HDDS projects. (Started to [be used](https://issues.apache.org/jira/browse/HDFS-11519) by Ozone in March of 2017)
```
Comment (dineshchitlangia) [inline HISTORY.md:33]:
```suggestion
In the beginning of the year 2017 a new podling project was started inside [Apache Incubator](http://incubator.apache.org/): [Apache Ratis](https://ratis.apache.org/). Ratis is an embeddable RAFT protcol implementation it is which became the corner stone of consensus inside both Ozone and HDDS projects. (Started to [be used](https://issues.apache.org/jira/browse/HDFS-11519) by Ozone in March of 2017) 
```

File: HISTORY.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+ > * HDSL become a subproject of Hadoop.
+ > * HDSL will release separately from Hadoop. Hadoop releases will notcontain HDSL and vice versa.
+ > * HDSL will get its own jira instance so that the release tags stay separate.
+ > * On trunk (as opposed to release branches) HDSL will be a separate module in Hadoop's source tree. This will enable the HDSL to work on their trunk and the Hadoop trunk without making releases for every change.
+ > * Hadoop's trunk will only build HDSL if a non-default profile is enabled. When Hadoop creates a release branch, the RM will delete the HDSL module from the branch.
+ > * HDSL will have their own Yetus checks and won't cause failures in the Hadoop patch check.
+
+This proposal is passed and after reorganizing the code (see HDFS-13258) and Ozone [has been voted](https://lists.apache.org/thread.html/ad0fe160ae84be97a0a87865059761ad7cd747be7b2fe060707d4f28@%3Chdfs-dev.hadoop.apache.org%3E) to be merged to the Hadoop trunk at the March of 2018.
```
Comment (dineshchitlangia) [inline HISTORY.md:44]:
```suggestion
This proposal was passed and after reorganizing the code (see HDFS-13258) and Ozone [has been voted](https://lists.apache.org/thread.html/ad0fe160ae84be97a0a87865059761ad7cd747be7b2fe060707d4f28@%3Chdfs-dev.hadoop.apache.org%3E) to be merged to the Hadoop trunk at the March of 2018.
```

File: HISTORY.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+After the merge a new Jira project was created (HDDS) and the work was tracked under that project instead of child issues under HDFS-7240.
+
+In the next year multiple Ozone releases has been published in separated release package. The Ozone source release was developed on the Hadoop trunk, but the Ozone sources are removed from the main Hadoop releases.
+
+Originally, Ozone depended on the in-tree (SNAPSHOT) Hadoop artifacts. It was required to compile the core hadoop-hdfs/hadoop-common artifacts before compiling the Ozone subprojects. During the development this dependency was reduced more and more. With the 0.4.1 release this dependency has been totally removed and it became possible to compile Ozone with the help of the released Hadoop artifacts which made it possible to separate the development of Ozone from the main Hadoop trunk branch.
+
+At the October of 2019 the Ozone sources are moved out to the [apache/hadoop-ozone](https://github.com/apache/hadoop-ozone) git reposity. During this move the git history has been transformed to remove old YARN/HDFS/MAPREDUCE tasks.
```
Comment (dineshchitlangia) [inline HISTORY.md:54]:
```suggestion
In October 2019, the Ozone sources were moved out to the [apache/hadoop-ozone](https://github.com/apache/hadoop-ozone) git repository. During this move the git history was transformed to remove old YARN/HDFS/MAPREDUCE tasks. 
```

**Pull request merged**